syntax = "proto3";

service MasterService {
  rpc FindLocations (FileRequest) returns (ChunkLocations) {}
  rpc GetFileLength (FileRequest) returns (FileLength) {}
  rpc CreateFile (FileRequest) returns (FileResponse) {}
  rpc DeleteFile (FileRequest) returns (FileResponse) {}
}

service ChunkServerService {
  rpc ReadChunk (ChunkRequest) returns (ChunkData) {}
  rpc WriteChunk (ChunkRequest) returns (ChunkResponse) {}
}

message FileRequest {
  string filename = 1;
}

message ChunkLocations {
  repeated string locations = 1;
}

message FileLength {
  int32 length = 1;
}

message FileResponse {
  string message = 1;
}

message ChunkRequest {
  string chunk_handle = 1;
  bytes data = 2;
}

message ChunkData {
  bytes data = 1;
}

message ChunkResponse {
  string message = 1;
}